vllm/v1/worker/gpu_input_batch.py

```diff
@@ -48,6 +48,13 @@ class CachedRequestState:
     def __post_init__(self):
         self.num_prompt_tokens = length_from_prompt_token_ids_or_embeds(
             self.prompt_token_ids, self.prompt_embeds)
+        # Establish the initial rollback checkpoint for sync-KV-load failures:
+        # capture the current output-token count (expected to be 0) and, if
+        # present, the generator’s internal state. If the request is later
+        # preempted, this checkpoint is refreshed when the request is resumed.
+        self.rollback_num_output_tokens = len(self.output_token_ids)
```

Comment:
Thinking aloud but wondering whether `original_num_output_tokens` or `starting_num_output_tokens` would be clearer in terms of what the value actually corresponds to.